src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -599,6 +631,52 @@ def __init__(self, children=None, parent=None, grainsize=None,
                 "numtasks clauses specified.")
         super(OMPTaskloopDirective, self).__init__(children=children,
                                                    parent=parent)
+        if self._grainsize is not None:
+            child = [Literal(f"{grainsize}", INTEGER_TYPE)]
+            self._children.append(GrainsizeClause(children=child))
+        if self._num_tasks is not None:
+            child = [Literal(f"{num_tasks}", INTEGER_TYPE)]
+            self._children.append(NumTasksClause(children=child))
+        if self._nogroup:
+            self._children.append(NogroupClause())
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+         Decides whether a given child and position are valid for this node.
+         The rules are:
+         1. Child 0 must always be a Schedule.
+         2. Child 1 may be either a GrainsizeClause or NumTasksClause, or if
+            neither of those clauses are present, it may be a NogroupClause.
+         3. Child 2 must always be a NogroupClause, and can only exist if child
+            1 is a GrainsizeClause or NumTasksClause
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        if position == 0:
+            return isinstance(child, Schedule)
+        if position == 1:
+            return isinstance(child, (GrainsizeClause, NumTasksClause,
+                                      NogroupClause))
+        if position == 2:
+            return (isinstance(child, NogroupClause))
+        return False
+
+    @property
+    def clauses(self):
+        '''
+        :returns: the Clauses associated with this directive.
+        :rtype: List of :py:class:`psyclone.psyir.nodes.Clause`
+        '''
+        if len(self.children) > 1:
+            return self.children[1:]
+        return []
```

Comment:
I think this should be in the super class RegionDirective. And a similar one in StandaloneDirective with just `return self.children`

Reply:
StandaloneDirective should return [] I think right? They have no clauses yet I think.